lab-michaelstuart/route/user-router.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+const Router = require('express').Router;
+const jsonParser = require('body-parser').json();
+const User = require('../model/user.js');
+
+let noteRouter = module.exports = new Router();
+
+noteRouter.post('/api/user', jsonParser, (req, res, next) => {
+  if (req.body.name && req.body.password) {
+    new User(req.body)
+      .save()
+      .then(user => res.json(user))
+      .catch(next);
+  } else {
+    res.sendStatus(404);
```

Comment:
I'm not sure if 404 is there error that you want to send back here. If the client doesn't supply required information it would be a 'bad request' instead of a 'not found'. Here, if you're checking for name/password it could be a 401 'unauthorized' rather than the generic 400.